enhancements/ingress/forwarded-header-policy.md

```diff
@@ -0,0 +1,325 @@
+	ReplaceHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Replace"
+	// IfNoneHTTPHeaderPolicy sets the header if it is not already set.
+	IfNoneHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "IfNone"
+	// NeverHTTPHeaderPolicy never sets the header, preserving any existing
+	// header.
+	NeverHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Never"
+)
+
+// IngressControllerHTTPHeaders specifies how the IngressController handles
+// certain HTTP headers.
+type IngressControllerHTTPHeaders struct {
+	// forwardedHeaderPolicy specifies when and how the IngressController
+	// sets the Forwarded, X-Forwarded-For, X-Forwarded-Host,
+	// X-Forwarded-Port, X-Forwarded-Proto, and X-Forwarded-Proto-Version
+	// HTTP headers.  The value may be one of the following:
+	//
+	// * "Append", which specifies that the IngressController appends the
+	//   headers, preserving existing headers.
+	//
+	// * "Replace", which specifies that the IngressController sets the
+	//   headers, replacing any existing Forwarded or X-Forwarded-* headers.
+	//
+	// * "IfNone", which specifies that the IngressController sets the
+	//   headers if they are not already set.
+	//
+	// * "Never", which specifies that the IngressController never sets the
+	//   headers, preserving any existing headers.
+	//
+	// By default, the policy is "Append".
+	//
+	// +optional
+	ForwardedHeaderPolicy IngressControllerHTTPHeaderPolicy `json:"forwardedHeaderPolicy,omitempty"`
+}
+```
+
+By default, the IngressController appends `Forwarded` and related headers to
+whatever headers are already present in the HTTP request.  The
+`ForwardedHeaderPolicy` field enables the user to modify this behavior, as
+described in the godoc above.  For example, the following IngressController
+replaces any `Forwarded` and related headers before forwarding the request:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  httpHeaders:
+    forwardedHeaderPolicy: Replace
+```
+
+Routes may override this policy using the
+`haproxy.router.openshift.io/set-forwarded-headers` annotation, which allows the
+values "append", "replace", "if-none", and "never", corresponding to the allowed
+values for the IngressController's `spec.httpHeaders.forwardedHeaderPolicy`
+field.  For example, an IngressController should never set the `Forwarded`
+header or related headers for the following Route, irrespective of the
+IngressController's own particular policy:
+
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  annotations:
+    haproxy.router.openshift.io/set-forwarded-headers: never
+  name: hello-openshift
+  namespace: hello-openshift
+spec:
+  to:
+    kind: Service
+    name: hello-openshift
+```
+
+#### Validation
+
+Omitting `spec.httpHeaders` or omitting `spec.httpHeaders.forwardedHeaderPolicy`
+specifies the default behavior.
```

Comment:
be precize. Is this defaulted in the CRD or is this just default behaviour of the controllers?